packages/core/src/utils/types.ts

```diff
@@ -37,3 +38,57 @@ export type Prettify<T> = {
 
 export type WithRequired<T, K extends keyof T> = Required<Pick<T, K>>;
 export type WithOptional<T, K extends keyof T> = Pick<Partial<T>, K>;
+
+// Checks for type equivalence, evaluates always to either `true` or `false`.
+// Semantics: (EQ<A, B> = true) <==> (A <==> B)
+export type EQ<A, B> = [A] extends [B]
+  ? [B] extends [A]
+    ? true
+    : false
+  : false;
+
+// Auxiliary type for `IsMemberOrSubtypeOfAComponent`
+// Evaluates to `true` or `boolean` if one of the explicit components (according to the TypeScript distribution
+// mechanism for union types) of the union type equals the other given type.
+// Examples:
+//    EqualsOneOfTheComponents<string, string | boolean> ⟶ boolean
+//    EqualsOneOfTheComponents<string, string> ⟶ true
+//    EqualsOneOfTheComponents<number, string | boolean> ⟶ false
+export type EqualsOneOfTheComponents<T, Union> = Union extends infer Component // enforce distribution
+  ? EQ<T, Component>
+  : never;
+
+// Auxiliary type for `IsOneOf`
+export type IsMemberOrSubtypeOfAComponent<
+  T,
+  Union,
+  ConjunctionOfExplicitComponentChecks extends boolean
+> = [T] extends [Union]
+  ? true extends ConjunctionOfExplicitComponentChecks
+    ? true
+    : false
+  : false;
+
+// Checks whether the given type equals one of the explicit components of the union type. Note that in this respect we
+// consider the members of the infinite types number and string as implicit since the inbuilt distribution mechanism of
+// TypeScript does not (cannot) distribute over the infinite number of strings or numbers. However, `boolean` is treated
+// by TypeScript (and accordingly here) as the explicit union `true | false`.
+//
+// In particular, the evaluation result is `false` if the type is a proper union of components of the union type. It also
+// evaluates to false if the given type is a proper subtype of one of its explicit components.
+// It always evaluates to either `true` or `false`.
+// Examples:
+//    IsOneOf<number, string | number | boolean> ⟶ true
+//    IsOneOf<2 | 4, 0 | 1 | 2 | 3 | 4 | 5 | 6> ⟶ false
+//    IsOneOf<'text', string> ⟶ false  // only implicit but not explicit component
+export type IsOneOf<T, Union> = IsMemberOrSubtypeOfAComponent<
+  T,
+  Union,
+  EqualsOneOfTheComponents<T, Union>
+>;
+
+export type OneOf<T1, T2> = IsOneOf<T1, T2> extends true ? T1 : never;
+
+export type Interface<T> = {
+  [K in keyof T]: any;
```

Comment:
should this actually be:
```suggestion
  [K in keyof T]: T[K];
```

Reply:
this type is removed as unused anymore. UserOperationLike is removed.

Reply:
bump, I still see it here?